### PR TITLE
docs: replicas

### PR DESCRIPTION
**Pull request by cristianmtr**
Closes https://github.com/jina-ai/internal-tasks/issues/74

Also converted the md to rst

### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [x] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [x] Are your section headers in sentence case?

### CODE REVIEW

File: chapters/parallel.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
+Parallelization
+---------------
+
+Learn how Jina handles parallelization of data processing
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
```
Comment (FionnD) [inline chapters/parallel.rst:6]: I think a table of contents would help

File: chapters/parallel.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+^^^^^^^^^^^^^^^^
+
+Another level of parallelism is within the Pods themselves, with **Replicas**.
+These operate inside a Pod, by creating duplicates of one Pod.
+The requests will then be sent to **one** of the Pods (the equivalent of ``polling: any``).
+In this case, a **Pod** is the **Replica** itself.
+Internally, Jina switches the class from the Pod to a CompoundPod, which contains multiple Pods.
+
+This feature was introduced for the `Querying while Indexing feature </chapters/dump-reload/>`_.
+
```
Comment (FionnD) [inline chapters/parallel.rst:72]: **Suggestion**: Some form of a diagram would be good.

File: chapters/parallel.rst [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+The plot of the above topology looks like this:
+
+|image0|
+
+In the above, ``p1``, ``p2``, ``p3`` now subscribe to ``gateway`` and
+conduct their work in parallel. The last ``.needs()`` blocks all Pods
+until they finish their work.
+
+Replicas of Pods
```
Comment (FionnD) [inline chapters/parallel.rst:58]: **Suggestion**: The advantages of using this approach over the above one would be good to know...

**Review by FionnD — CHANGES_REQUESTED** [review]
Great work, some minor comments. The table of contents is a requirement, the other two comments are suggestions.

File: chapters/parallel.rst [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+---------------
+
+Learn how Jina handles parallelization of data processing
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Feature description and expected outcome
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Jina allows for parallelization of Peas and Pods. This generally leads
+to an increase in throughput, as the data is split across processes.
```
Comment (theUnkownName) [inline chapters/parallel.rst:11]: typo "throughput"?

File: chapters/parallel.rst [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+You are expected to be familiar with the general concepts of Jina, as
+presented in
+`here <https://docs.jina.ai/chapters/core/introduction/index.html>`__.
+
+You should also have experience with parallelism in other systems. Jina
+does follow the best practices and designs from other databases or
+search engines. However, Jina is primarily a *neural search engine*.
+This means some changes and adaptations were necessary. This chapter
+will explain these.
```
Comment (CatStark) [inline chapters/parallel.rst]: Try to use active voice

File: chapters/parallel.rst (hunk 4 above)
Comment (CatStark) [inline chapters/parallel.rst]: Maybe change "throughput" for a simpler word

File: chapters/parallel.rst [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+until they finish their work.
+
+Replicas of Pods
+^^^^^^^^^^^^^^^^
+
+Another level of parallelism is within the Pods themselves, with **Replicas**.
+These operate inside a Pod, by creating duplicates of one Pod.
+The requests will then be sent to **one** of the Pods (the equivalent of ``polling: any``).
+In this case, a **Pod** is the **Replica** itself.
+Internally, Jina switches the class from the Pod to a CompoundPod, which contains multiple Pods.
```
Comment (CatStark) [inline chapters/parallel.rst]: Can you also highlight `CompoundPod`please?

File: chapters/parallel.rst [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+This feature was introduced for the `Querying while Indexing feature </chapters/dump-reload/>`_.
```
Comment (CatStark) [inline chapters/parallel.rst]: Let's try to use active voice

File: chapters/parallel.rst [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+In order to enable this feature for a specific Pod add the following to your YAML or Python programmatic API:
```
Comment (CatStark) [inline chapters/parallel.rst]: Try to breakdown this sentence

File: chapters/parallel.rst [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+**Limitations**:
+
+1. Replicas always have ``polling: any``.
+2. Replicas are currently running on the same machine. We are working on being able to separate them to distinct machines.
+
+Intra-parallelism
+^^^^^^^^^^^^^^^^^
+
+Peas are the next level at which we can configure parallelism. In this
```
Comment (CatStark) [inline chapters/parallel.rst:98]: Highlight the first occurrence of a Jina keyword (Peas)

File: chapters/parallel.rst [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+              .needs(['p1','p3'], name='r1').plot())
+
+|image1|
+
+State-less vs state-ful
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Depending on the type of ``Peas``, ``parallel`` means different things.
+In *state-less* ``Peas`` (``Crafters``, ``Encoders`` etc.), you
+*parallelize*, not *shard*. This means the data is not persisted, but
```
Comment (CatStark) [inline chapters/parallel.rst:117]: Stick to one type of highligting

File: chapters/parallel.rst [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+seen by all the ``Peas``, and that the respective ``id`` is found. If we
+use ``polling: any`` we risk that the key is not seen by the shard that
+has the data.
+
+Scheduling
+^^^^^^^^^^
+
+Jina offers two types of scheduling:
+
+1. ``round_robin`` will assign tasks in circular order, without any
```
Comment (CatStark) [inline chapters/parallel.rst]: Let's use active voice

File: chapters/parallel.rst [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+   prioritization. We assume all ``Peas`` have uniform processing power.
+
+2. ``load_balance`` will assign tasks to the idle ``Peas``.
```
Comment (CatStark) [inline chapters/parallel.rst]: Active voice here too

File: chapters/parallel.rst [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+.. code:: python
+
+   from jina.enums import  SchedulerType
+   f = Flow().add(name='p1', parallel=3, scheduling=SchedulerType.ROUND_ROBIN)
+
+Limitations
+~~~~~~~~~~~
+
+At the moment Jina does not support sharding and replication of shards
```
Comment (CatStark) [inline chapters/parallel.rst]: Rephrase a bit this part

**Comment by cristianmtr** [comment]
@theUnkownName @FionnD I've implemented your requested changes. Please check again :)

File: chapters/parallel.rst [hunk 14]
```diff
@@ -0,0 +1,166 @@
+Parallelization
+---------------
```
Comment (nan-wang) [inline chapters/parallel.rst]: The section levels need to be increased by one level
```suggestion
===========
```

File: chapters/parallel.rst [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+An example of configuring the above parameters can be seen here:
+
+.. code:: python
+
+   from jina.enums import  SchedulerType
+   f = Flow().add(name='p1', parallel=3, scheduling=SchedulerType.ROUND_ROBIN)
+
+
+.. |image0| image:: ./images/simple-plot3.svg
+.. |image1| image:: ./images/simple-plot4.svg
```
Comment (nan-wang) [inline chapters/parallel.rst:165]: the related images are missing in this PR.

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Comment by cristianmtr** [comment]
@nan-wang Bumped the header level.

I don't get your comment about the images. They render fine for me locally.

**Review by CatStark — APPROVED** [review]
LGTM!

File: chapters/parallel.rst [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+~~~~~~~~~~~~~~~~
+
+You are expected to be familiar with the general concepts of Jina, as
+presented in
+`here <https://docs.jina.ai/chapters/core/introduction/index.html>`__.
+
+You should also have experience with parallelism in other systems. Jina
+does follow the best practices and designs from other databases or
+search engines. However, Jina is primarily a *neural search engine*.
+This means some changes and adaptations were necessary. I will explain these here.
```
Comment (FionnD) [inline chapters/parallel.rst]: Remove: I will explain these here.

File: chapters/parallel.rst [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+until they finish their work.
+
+Replicas of Pods
+^^^^^^^^^^^^^^^^
+
+Another level of parallelism is within the Pods themselves, with **Replicas**.
+These operate inside a Pod, by creating duplicates of one Pod.
+This is different from the inter-parallelism provided by the ``needs`` parameter,
+as that one operates between different Pods of different business logic.
+On the other hand, the Replicas operate within
```
Comment (FionnD) [inline chapters/parallel.rst]: Fullstop

**Review by FionnD — APPROVED** [review]
Please fix the two comments before merging

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Comment by nan-wang** [comment]
> @nan-wang Bumped the header level.
> 
> I don't get your comment about the images. They render fine for me locally.

My bad. The images are there. All good.

**Pull request merged**